Userland/Libraries/LibJS/Runtime/VM.cpp

```diff
@@ -193,8 +193,11 @@ Value VM::get_variable(const FlyString& name, GlobalObject& global_object)
 
         for (auto* scope = current_scope(); scope; scope = scope->parent()) {
             auto possible_match = scope->get_from_scope(name);
-            if (possible_match.has_value())
+            if (possible_match.has_value()) {
                 return possible_match.value().value;
+            } else if (exception()) {
+                return {};
+            }
```

Comment:
I'd prefer the usual "exception check first, return immediately if any" style:

```suggestion
            if (exception())
                return {};
            if (possible_match.has_value())
                return possible_match.value().value;
```

Reply:
Ah gotcha, thanks for the tip!